packages/legend-query-builder/src/components/fetch-structure/QueryBuilderTDSPanel.tsx

```diff
@@ -484,12 +486,12 @@ const QueryBuilderProjectionColumnEditor = observer(
       (type: string): void => {
         if (
           type === QUERY_BUILDER_PROJECTION_COLUMN_DND_TYPE &&
-          tdsState.draggedColumnIndex !== undefined &&
-          tdsState.hoveredColumnIndex !== undefined
+          tdsState.hoveredColumnIndex !== undefined &&
+          tdsState.draggedColumnIndex !== undefined
```

Comment:
let's do the math here to figure out the `swap` column idx based from the gap index`